### PR TITLE
[runtime-security] safer zero event

### PR DESCRIPTION
**Pull request by safchain**
### What does this PR do?

Revert previous zero event approach and introduce a safer one.

### Motivation

It is safer to zero the whole structure at once than zero field by field depending on the event type, some field could be missing because not handled by the generator. 

### Additional Notes

Anything else we should know when reviewing?

### Describe your test plan

Write there any instructions and details you may have to test your PR.

### CODE REVIEW

**Comment by tanguylebarzic** [comment]
nit: would be nice to explain how this new version is safer, for the benefits of current and future readers :).

**Comment by safchain** [comment]
@tanguylebarzic I updated to PR comment

**Review by guedou — APPROVED** [review]

File: pkg/security/probe/probe_bpf.go [hunk 1]
```diff
@@ -258,11 +267,7 @@ func (p *Probe) handleEvent(CPU int, data []byte, perfMap *manager.PerfMap, mana
 	offset += read
 
 	eventType := EventType(event.Type)
-
-	// zero the event first
-	event.Zero(eventType.String())
-
-	log.Tracef("Decoding event %s", eventType)
+	log.Tracef("Decoding event %s", eventType.String())
```
Comment (Gui774ume) [inline pkg/security/probe/probe_bpf.go:270]: nit: isn't the `%s` format already making sure that the `String` function is called ?

**Review by Gui774ume — APPROVED** [review]
LGTM :+1:

File: pkg/security/probe/probe_bpf.go (hunk 1 above)
Comment (safchain) [reply pkg/security/probe/probe_bpf.go]: it is re-added by the revert :) anyway I fixed it in the following commit

**Pull request merged**